images/devcontainer/features.md

```diff
@@ -82,8 +82,12 @@ Several Dev Container Features allow R package installation to be defined on `de
 
 These support package installation via `apt` or R function ([`pak::pak()`](https://pak.r-lib.org/reference/pak.html)).
 
-There is also [the `renv-cache` Feature](https://github.com/rocker-org/devcontainer-features/tree/main/src/renv-cache)
-that supports package installation via [the `renv` R package](https://rstudio.github.io/renv/) after container startup.
+There are also [`r-dependent-packages`](https://github.com/rocker-org/devcontainer-features/tree/main/src/r-dependent-packages)
+and [`renv-cache`](https://github.com/rocker-org/devcontainer-features/tree/main/src/renv-cache).
+
+- `r-dependent-packages` supports package installation from a `DESCRIPTION` file via
+  ([`pak::local_install_deps()`](https://pak.r-lib.org/reference/local_install_deps.html)) after container startup.
+- `renv-cache` supports package installation via [the `renv` R package](https://rstudio.github.io/renv/) after container startup.
```

Comment:
Out of curiousity:  Why do use pak when you could use r2u?  Some of us (eg Grant) had good luck with r2u in devcontainers; he promoted that at work too AFAIK.

Reply:
My understanding is that r2u can only be used in limited situations:

- Does not support arbitrary CRAN URLs
- Only amd64 Ubuntu is supported

I expect that a version using r2u could be created by using something like `remotes::install_deps()` instead of `pak::local_install_deps()`, but I don't know if there is demand for it.

Reply:
For example, this is an example I am actually using, but this set an additional repository and installed polars from R-universe, so perhaps bspm can't handle this?

<https://github.com/eitsupi/r-glaredb/blob/5a5d9c3564c763e988492b142d54d2c635e5b4f3/.devcontainer/devcontainer.json#L8-L12>

```json
"ghcr.io/rocker-org/devcontainer-features/r-dependent-packages": {
	"when": "updateContent",
	"additionalRepositories": "polars = 'https://rpolars.r-universe.dev'",
	"dependencyTypes": "all,Config/Needs/dev,Config/Needs/website"
},
```

Reply:
That is incorrect.  Just give it a try.

We (at work) use install deps from remote all the time with r2u in CI and other settings.  I use it all the time to debug a repo.  It gets all it can via apt, what it cannot ("Additional_repositories") it still installs the usual way.  I personally use it via littler wrappers installDeps.r and installDeps2.r (a bit like the install.r and install2.r story: eventually I wanted more options).  In short, because Inaki did the Right Thing (TM) tracing install.packages is still works here.

I also use things like polars from r-universe and have a wrapper installRub.r for that. I am unsure what remotes does here so I usually just combine the two with two calls:  installRub.r for what I want from r-universe, standard remotes::install-deps() in lieu of install.packages.  Happy to demo.

(Nothing wrong with pak.  But _for Ubuntu_ as we often use for containers etc I feel r2u (fairly clearly) dominates.  Speed is one thing, the better logic from integrating with the system package manager is another.  But tastes vary so :man_shrugging: )

Reply:
Okay, thanks.
If that's the case, it might be worth adding an option to use `remotes::install_deps()` as the function for installation.

Reply:
I actually wrote (based on noodling with Grant) a post on this for r2u and devcontainers with r-universe:
https://dirk.eddelbuettel.com/blog/2023/08/15/#041_r2u_and_codespaces